File: repos_estudiantes_ALVAREZ_segunda_entrega/ILLANES_VASQUEZ_GABRIEL/Practica2prog/Composicion y Agregacion/Ejer 9/CarritoCompras.java
package Agr9;

public class CarritoCompras {
    private int nroProd;
    private Producto productos[] = new Producto[10];

    public CarritoCompras() {
        nroProd = 0;
    }

    public void mostrar_carrito() {
        System.out.println("Cantidad de productos: " + nroProd);
        for (int i = 0; i < nroProd; i++) {
            productos[i].mostrar_info();
        }
    }

    public void agregar_producto(Producto x) {
        if (nroProd < 10) {
            productos[nroProd] = x;
            nroProd++;
        } else {
            System.out.println("Ya tiene 10 Productos en su carrito :c, ese es el límite");
            System.out.println("No se agregará");
            x.mostrar_info();
        }
    }
}

File: repos_estudiantes_TARQUI_segunda_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/AGREGACION/EJERCICIO-9/CarritoCompras.py
from Producto import Producto

class CarritoCompras:
    MAX_PRODUCTOS = 10
    
    def __init__(self):
        self.__productos = []
    
    def agregar_producto(self, producto: Producto) -> bool:
        if len(self.__productos) >= self.MAX_PRODUCTOS:
            print("¡Error! El carrito no puede contener más de 10 productos")
            return False
        self.__productos.append(producto)
        return True
    
    def mostrar_carrito(self) -> None:
        print("\n--- Contenido del Carrito ---")
        for producto in self.__productos:
            producto.mostrar_info()
        print(f"Total productos: {len(self.__productos)}/{self.MAX_PRODUCTOS}")
        print("----------------------------")

    def get_productos(self) -> list[Producto]:
        return self.__productos